### PR TITLE
Support priority on Mail.Options

### PR DESCRIPTION
**Pull request by poseidonCore**
As stated in this file:
https://github.com/nodemailer/nodemailer/blob/master/lib/mailer/mail-message.js

### CODE REVIEW

File: types/nodemailer/lib/mailer/index.d.ts [hunk 1]
```diff
@@ -137,6 +137,7 @@ declare namespace Mail {
         dkim?: DKIM.Options;
         /** method to normalize header keys for custom caseing */
         normalizeHeaderKey?(key: string): string;
+        priority?: "high"|"normal"|"low";
```
Comment (bioball) [inline types/nodemailer/lib/mailer/index.d.ts]: Seems like this should just be `"high"` or `"low"`, based off their source code

https://github.com/nodemailer/nodemailer/blob/master/lib/mailer/mail-message.js#L225...L238

**Review by bioball — CHANGES_REQUESTED** [review]
Didn't see a `"normal"` priority anywhere?

**Comment by poseidonCore** [comment]
@biobal: ... It is the default, as in the code you quoted ...

https://github.com/nodemailer/nodemailer/blob/master/lib/mailer/mail-message.js#L236...L237

**Comment by bioball** [comment]
When is the "normal" value actually used? It doesn't seem like that means anything in code.

**Comment by poseidonCore** [comment]
It's used in cases where programmers are explicitly specifying the priority, rather than just assuming it.

If 'normal' is passed, then it means in the code that no priority is set in the email itself, which its way of implying 'normal' priority, as described in the code itself : `// do not add anything, since all messages are 'Normal' by default`.

The fact that the code does nothing with the value does not negate its semantic significance to the consumer of the type interface. 

It is obvious that the code accommodates this as an option, and therefore it should be an option. Otherwise, it is misleading to say that the only priorities are just  `"high"|"low"`.

**Comment by bioball** [comment]
Ok, that makes sense to me

**Review by bioball — APPROVED** [review]
LGTM

**Pull request merged**